solana/pyth2wormhole/client/src/main.rs

```diff
@@ -225,181 +245,307 @@ async fn main() -> Result<(), ErrBox> {
 
     Ok(())
 }
 
-/// Send a series of batch attestations for symbols of an attestation config.
-async fn handle_attest(
-    rpc_url: String,
-    commitment: CommitmentConfig,
+/// Continuously send batch attestations for symbols of an attestation config.
+async fn handle_attest_daemon_mode(
+    rpc_cfg: Arc<RLMutex<RpcCfg>>,
     payer: Keypair,
     p2w_addr: Pubkey,
-    attestation_cfg: AttestationConfig,
-    n_retries: usize,
-    retry_interval: Duration,
-    confirmation_timeout: Duration,
-    daemon: bool,
+    mut attestation_cfg: AttestationConfig,
 ) -> Result<(), ErrBox> {
-    // Derive seeded accounts
-    let emitter_addr = P2WEmitter::key(None, &p2w_addr);
-
-    info!("Using emitter addr {}", emitter_addr);
-
-    let config = get_config_account(
-        &RpcClient::new_with_timeout_and_commitment(
-            rpc_url.clone(),
-            confirmation_timeout,
-            commitment.clone(),
-        ),
-        &p2w_addr,
-    )
-    .await?;
+    info!(
+        "Crawling mapping {:?} every {} minutes",
+        attestation_cfg.mapping_addr, attestation_cfg.mapping_reload_interval_mins
+    );
 
-    debug!("Symbol config:\n{:#?}", attestation_cfg);
+    let mut old_sched_futs_state: Option<(JoinHandle<_>, u64)> = None; // (old_futs_handle, old_config_hash)
 
-    info!(
-        "{} symbol groups read, dividing into batches",
-        attestation_cfg.symbol_groups.len(),
+    // This loop cranks attestations without interruption. This is
+    // achieved by spinning up a new up-to-date symbol set before
+    // letting go of the previous one. Additionally, hash of on-chain
+    // and attestation configs is used to prevent needless reloads of
+    // an unchanged symbol set.
+    loop {
+        let start_time = Instant::now(); // Helps timekeep mapping lookups accurately
+
+        let config = get_config_account(&lock_and_make_rpc(&rpc_cfg).await, &p2w_addr).await?;
+
+        // Use the mapping if specified
+        if let Some(mapping_addr) = attestation_cfg.mapping_addr.as_ref() {
+            match crawl_pyth_mapping(&lock_and_make_rpc(&rpc_cfg).await, mapping_addr).await {
+                Ok(additional_accounts) => {
+                    debug!(
+                        "Crawled mapping {} data:\n{:#?}",
+                        mapping_addr, additional_accounts
+                    );
+                    attestation_cfg.add_symbols(additional_accounts, "mapping".to_owned());
+                }
+                // De-escalate crawling errors; A temporary failure to
+                // look up the mapping should not crash the attester
+                Err(e) => {
+                    error!("Could not crawl mapping {}: {:?}", mapping_addr, e);
+                }
+            }
+        }
+        debug!(
+            "Attestation config (includes mapping accounts):\n{:#?}",
+            attestation_cfg
+        );
+
+        // Hash currently known config
+        let mut hasher = DefaultHasher::new();
```

Comment:
i think i left a comment last time about using a cryptographic hash here. The code below depends on no collisions for correctness.

Reply:
That makes sense. I think the std HashMap has internal collision detection that isn't part of the hasher. I assumed it would be enough, but a nicer 32-byte scheme will fit better here.